service/service.go

```diff
@@ -165,26 +165,33 @@ func (t *service) Store() lstore.Logstore {
 	return t.store
 }
 
-// GetHostID returns the host's peer ID.
+// GetHostID returns the host's peer id.
 func (t *service) GetHostID(context.Context) (peer.ID, error) {
 	return t.host.ID(), nil
 }
 
 // CreateThread with id.
-func (t *service) CreateThread(_ context.Context, id thread.ID) (info thread.Info, err error) {
-	info.ID = id
-	info.FollowKey, err = sym.CreateKey()
-	if err != nil {
-		return
+func (t *service) CreateThread(_ context.Context, id thread.ID, opts ...core.KeyOption) (info thread.Info, err error) {
+	args := &core.KeyOptions{}
+	for _, opt := range opts {
+		opt(args)
 	}
-	info.ReadKey, err = sym.CreateKey()
-	if err != nil {
-		return
+
+	info = thread.Info{
+		ID:        id,
+		FollowKey: args.FollowKey,
+		ReadKey:   args.ReadKey,
+	}
+	if info.FollowKey == nil {
```

Comment:
If there's no follow key given with `CreateThread`, the service will create one for you.

Reply:
Aha, ok this answers my pervious question I think...